lib/src/date-input/Calendar.tsx

```diff
@@ -210,15 +228,24 @@ const WeekHeaderCell = styled.span`
   height: 36px;
 `;
 
-const DayCellsContainer = styled.div`
+const DayCellContainer = styled.span``;
```

Comment:
DayCellContainer is no longer used.

Reply:
As you proposed, the container with the `gridcell` role should not be needed, the error thrown by axe-core is listed as a false positive (https://github.com/microsoft/fluentui/issues/30305#issuecomment-1889892108) So I added a custom rule for Date Input and added the role directly to the button.